doozerlib/runtime.py

```diff
@@ -983,7 +1000,7 @@ def resolve_source(self, meta):
         # This allows passing `--source <distgit_key> path` to
         # override any source to something local without it
         # having been configured for an alias
-        if self.local and meta.distgit_key in self.source_paths:
+        if self.local and meta.distgit_key in self.source_resolutions:
```

Comment:
As I look through this, it seems that at some point, resolve_source was designed to be single threaded. There are a number of race conditions I see now (introduced not by your PR, but by the use of these methods in a muli-threaded context). Your PR is subject to those pre-existing races. Soo.. what do you think about putting this item on the backlog until we can fully revisit threading in resolve_source?

Reply:
Sounds good to me. I'll log this to the backlog.